### PR TITLE
Loads the statistic subject structure xml file in a thread.

### PR DESCRIPTION
**Pull request by Jorgen-5**
- Made the fetch_statistcal_structure method run as a thread when starting the App.
- The _parse_statistic_subject_structure_xml now runs when get_primary_subject is called.
- Changed the tests to use the wait_for_primary_subject function to be able to parse the xml file in the tests.

### CODE REVIEW

File: src/datadoc/state.py [hunk 1]
```diff
@@ -20,3 +23,7 @@
 metadata: DataDocMetadata
 
 current_metadata_language: SupportedLanguages
+
+SOURCE_URL = "https://www.ssb.no/xp/_/service/mimir/subjectStructurStatistics"
```
Comment (mmwinther) [inline src/datadoc/state.py:27]: We shouldn't hard code this, but instead get it from the config, we added functionality for that in #158

File: src/datadoc/state.py [hunk 2]
```diff
@@ -20,3 +23,7 @@
...
+
+statistic_subjec_mapping = StatisticSubjectMapping(SOURCE_URL)
```
Comment (mmwinther) [inline src/datadoc/state.py:29]:
```suggestion
statistic_subject_mapping = StatisticSubjectMapping(SOURCE_URL)
```

Just a little typo

**Review by mmwinther — COMMENTED** [review]
Looks great, just two small comments

File: src/datadoc/backend/statistic_subject_mapping.py [hunk 3]
```diff
@@ -39,14 +40,16 @@ def __init__(self, source_url: str) -> None:
         self.statistic_short_name_secondary_subject_mapping: dict[str, str] = {
```
Comment (mmwinther) [inline src/datadoc/backend/statistic_subject_mapping.py:40]: We should remove lines 39-42, they're left over from my original experimentation.

Comment (Jorgen-5) [reply src/datadoc/backend/statistic_subject_mapping.py]: statistic_short_name_secondary_subject_mapping is used in the get primary subjects, but it seems that it is never updated, is this meant to be done like this or is this also code from the original experimentation?

**Pull request merged**